lib/hash_ring/ets.ex

```diff
@@ -41,202 +59,220 @@ defmodule ExHashRing.HashRing.ETS do
-  end
-
-  @spec set_nodes(atom, [binary | {binary, integer}]) :: {:ok, [{binary, integer}]}
-  def set_nodes(name, node_names) do
-    GenServer.call(name, {:set_nodes, node_names})
-  end
-
+  @doc """
+  Adds a node to the existing set of nodes in the ring.
+  """
   @spec add_node(atom, binary, integer) :: {:ok, [{binary, integer}]} | {:error, :node_exists}
   def add_node(name, node_name, num_replicas \\ nil) do
     GenServer.call(name, {:add_node, node_name, num_replicas})
   end
 
-  @spec remove_node(atom, binary) :: {:ok, [{binary, integer}]} | {:error, :node_not_exists}
-  def remove_node(name, node_name) do
-    GenServer.call(name, {:remove_node, node_name})
+  @doc """
+  Adds multiple nodes to the existing set of nodes in the ring.
+  """
+  @spec add_nodes(name :: atom, nodes :: [binary | {binary, integer}]) :: {:ok, [{binary, integer}]} | {:error, :node_exists}
+  def add_nodes(name, nodes) do
+    GenServer.call(name, {:add_nodes, nodes})
   end
 
-  @spec set_overrides(atom, override_map) :: {:ok, override_map}
-  def set_overrides(name, overrides) do
-    GenServer.call(name, {:set_overrides, overrides})
+  @doc """
+  Finds the node responsible for the given key in the specified ring.
+  """
+  @spec find_node(atom, binary | integer) :: {:ok, binary} | {:error, atom}
+  def find_node(name, key) do
+    with {:ok, [node]} <- find_nodes(name, key, 1)  do
+      {:ok, node}
+    end
   end
 
-  @spec get_overrides(atom) :: {:ok, override_map}
-  def get_overrides(name) do
-    GenServer.call(name, :get_overrides)
+  @doc """
+  Finds the specified number of nodes responsible for the given key in the specified ring's current generation.
+  """
+  @spec find_nodes(atom, binary | integer, integer) :: {:ok, [binary]} | {:error, atom}
+  def find_nodes(name, key, num) do
+    with {:ok, {{table, num_nodes}, _previous_table, gen, overrides}} <- Config.get(name) do
+      do_find_nodes_in_table(key, table, overrides, gen, num_nodes, num)
+    end
   end
 
-  @spec get_nodes(atom) :: {:ok, [binary]}
-  def get_nodes(name) do
-    GenServer.call(name, :get_nodes)
+  @doc """
+  Finds the specified number of nodes responsible for the given key in the specified ring's previous generation.
+  """
+  @spec find_previous_nodes(atom, binary | integer, integer) :: {:ok, [binary]} | {:error, atom}
+  def find_previous_nodes(name, key, num) do
+    with {:ok, {_current_table, {table, num_nodes}, gen, overrides}} <- Config.get(name) do
+      do_find_nodes_in_table(key, table, overrides, gen, num_nodes, num)
+    end
   end
 
-  @spec get_nodes_with_replicas(atom) :: {:ok, [{binary, integer}]}
-  def get_nodes_with_replicas(name) do
-    GenServer.call(name, :get_nodes_with_replicas)
+  @doc """
+  Finds the specified number of nodes responsible for the given key in the specified ring's current generation AND in
+  the specified ring's previous generation.  This means that this function returns up to 2 * number of nodes requested.
```

Comment:
nit: use the parameter you are referencing
```suggestion
  the specified ring's previous generation.  This means that this function returns up to 2 * `num`; where `num` = number of nodes.
```

could also just change the variable to something a bit more descriptive: `num_nodes`

Reply:
Adopted in https://github.com/discord/ex_hash_ring/pull/10/commits/173fbda6fff6a11075af73ad466eed548991667b